### PR TITLE
Guest OS Support: add JeOS/Fedora 22/23 

### PR DESCRIPTION
**Pull request by clebergnu**
This PR adds (late) support for Fedora 22 and 23, and also the JeOS guests based on it.

Newer JeOS images, based on this configuration/kickstart will be pushed to the official asset repo.

### CODE REVIEW

File: shared/cfg/guest-os/Linux/Fedora/22.x86_64.cfg [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        syslog_server_proto = tcp
+    unattended_install.cdrom, svirt_install:
+        cdrom_cd1 = isos/linux/Fedora-Server-DVD-x86_64-22.iso
+        md5sum_cd1 = 18165e85fbc9e975ec68631b252c9265
+        md5sum_1m_cd1 = d4803c6d5ffdccab9a6974bb09a1eb10
+    unattended_install.url:
+        kernel_params = 'ks=cdrom nicdelay=60 console=ttyS0,115200 console=tty0'
+        url = http://dl.fedoraproject.org/pub/fedora/linux/releases/22/Server/x86_64/os
+        sha1sum_vmlinuz = c0056a832df068e985e603c59f43efb2d5500336
+        sha1sum_initrd = 02fa64428e37cb137561a982d246468fe65f6e8d
```
Comment (ldoktor) [inline shared/cfg/guest-os/Linux/Fedora/22.x86_64.cfg]: Do I have to verify the `sha` sums, or should I believe you? :wink:

File: shared/cfg/guest-os/Linux/Fedora/22.x86_64.cfg [hunk 2]
```diff
@@ -0,0 +1,22 @@
+- 22.x86_64:
```
Comment (ldoktor) [inline shared/cfg/guest-os/Linux/Fedora/22.x86_64.cfg]: I'm wondering if we should add `armhfp`, but I'm not aware of anybody using it so we can probably wait for user request.

File: shared/unattended/Fedora-22.ks [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+bootloader --location=mbr --append="console=tty0 console=ttyS0,115200"
+zerombr
+poweroff
+KVM_TEST_LOGGING
+
+clearpart --all --initlabel
+autopart
+
+%packages
+@standard
```
Comment (ldoktor) [inline shared/unattended/Fedora-22.ks]: The `@development-tools` are missing from `Fedora-21`, is it intentional? Isn't it needed by some tests? (like `autotest` tests which need `gcc`?)

File: shared/cfg/guest-os/Linux/JeOS/22.x86_64.cfg [hunk 4]
```diff
@@ -0,0 +1,25 @@
+- 22.x86_64:
```
Comment (ldoktor) [inline shared/cfg/guest-os/Linux/JeOS/22.x86_64.cfg]: Honestly I don't think we need to release JeOS for each existing fedora. We had `JeOS.19`, now `JeOS.21` so I'd skip the `JeOS.22` as nobody needed it yet and now he can use `JeOS.23`...

File: shared/cfg/guest-os/Linux/Fedora/23.x86_64.cfg [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        syslog_server_proto = tcp
+    unattended_install.cdrom, svirt_install:
+        cdrom_cd1 = isos/linux/Fedora-Server-DVD-x86_64-23.iso
+        md5sum_cd1 = 04c9146c3a6cc86cf38e1f8a3b32865b
+        md5sum_1m_cd1 = 4d6c41d6935dc1962123c44cadeafd45
+    unattended_install.url:
+        kernel_params = 'ks=cdrom nicdelay=60 console=ttyS0,115200 console=tty0'
+        url = http://dl.fedoraproject.org/pub/fedora/linux/releases/23/Server/x86_64/os
+        sha1sum_vmlinuz = c28e39a33f7fdb61dcd7aac23b768ddf0826c925
+        sha1sum_initrd = eef2879e6107f10a013dcfa9c98da1d09a24c6e7
```
Comment (ldoktor) [inline shared/cfg/guest-os/Linux/Fedora/23.x86_64.cfg]: I haven't checked the `sha`s but apart from that it looks good.

File: shared/unattended/Fedora-23.ks [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+bootloader --location=mbr --append="console=tty0 console=ttyS0,115200"
+zerombr
+poweroff
+KVM_TEST_LOGGING
+
+clearpart --all --initlabel
+autopart
+
+%packages
+@standard
```
Comment (ldoktor) [inline shared/unattended/Fedora-23.ks]: The same as in 22, how about the missing `@development-tools`?

File: shared/unattended/JeOS-23.ks [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+bootloader --location=mbr --append="console=tty0 console=ttyS0,115200"
+zerombr
+poweroff
+KVM_TEST_LOGGING
+
+clearpart --all --initlabel
+part / --fstype=ext4 --grow --asprimary --size=1
+
+%packages
+gpgme
```
Comment (ldoktor) [inline shared/unattended/JeOS-23.ks]: Compare to `JeOS-21` this does not contain `hardlink` and `patch`. Are you 100% sure they are not needed for all the tests we have marked as `JeOS`-friendly? Or is it a dependency in F23?

File: shared/cfg/guest-os/Linux/JeOS/22.x86_64.cfg (hunk 4 above)
Comment (ldoktor) [inline shared/cfg/guest-os/Linux/JeOS/22.x86_64.cfg]: Oh we had `JeOS.20` I just never used it... Still I think we don't have to release each version.

**Comment by ldoktor** [comment]
@clebergnu so you want to first merge this and then build the `JeOS` and push the `jeos-23-64.ini` file when it's uploaded, is that correct?

File: shared/unattended/Fedora-22.ks (hunk 3 above)
Comment (clebergnu) [inline shared/unattended/Fedora-22.ks]: It's not available in the "Server" spin. I thought of changing the ISO/URL to "Workstation", but seemed adequate. At least to the subset of tests I ran with those guests.

File: shared/cfg/guest-os/Linux/JeOS/22.x86_64.cfg (hunk 4 above)
Comment (clebergnu) [inline shared/cfg/guest-os/Linux/JeOS/22.x86_64.cfg]: I don't plan to release it. I'm pushing JeOS.23 now, but not JeOS.22. Still, it doesn't hurt to help users generate their own, if they want to.

File: shared/unattended/Fedora-23.ks (hunk 6 above)
Comment (clebergnu) [inline shared/unattended/Fedora-23.ks]: Same issue with the "Server" spin.

File: shared/unattended/JeOS-23.ks (hunk 7 above)
Comment (clebergnu) [inline shared/unattended/JeOS-23.ks]: Not available as well. I run a base set of tests, and they ran OK.

**Comment by clebergnu** [comment]
Correct. Actually, I'm done with it all, including the hosting part of the JeOS.23. I can push those commits to this same PR, or wait for this to be merged. What do you prefer?

File: shared/cfg/guest-os/Linux/JeOS/22.x86_64.cfg (hunk 4 above)
Comment (ldoktor) [inline shared/cfg/guest-os/Linux/JeOS/22.x86_64.cfg]: OK, btw after creating the JeOS.23 you should probably also update the `virttest/defaults.py`...

**Comment by ldoktor** [comment]
Separate PR would be IMO better, I'm just waiting for the results of the `autotest` tests to see if gcc is present...

**Comment by clebergnu** [comment]
Closing in favor of #439

**Comment by yashkmankad** [comment]
@clebergnu Thanks for this! 
JeOS 21 was causing a lot of issues. Me and @xutian  were discussing the need of JeOS based on F22/23. I ll fetch them from your repo. 
Thank you!

**Pull request closed without merging**